classes/question/bank/question_bank_filter.php

```diff
@@ -458,12 +479,12 @@ class user_filter_percent extends user_filter_number {
      * @return array
      */
     public function get_sql_filter($data) {
-        $val = round($data->value, 0);
+        $val = round($data['value'], 0);
         if ($val > 100 or $val < 0) {
             return '';
         }
         if ($val > 1) {
-            $data->value = $val / 100;
+            $data['value'] = $val / 100;
```

Comment:
If $data is an array, please update the PHPdoc to say that.

(Although, acutally, this is an overridden method, so the PHPdoc comment should be deleted.)

Reply:
Except that, it seems that in StudentQuiz, the consistently document overridden methods, to keep moodle-plugin-ci passing. So stick to that policy and keep the comment and fix it.